packages/dds/merge-tree/src/partialLengths.ts

```diff
@@ -650,22 +885,56 @@ export class PartialSequenceLengths {
 				const branchPartialLengths = childBlock.partialLengths!;
 				const partialLengths = branchPartialLengths.partialLengths;
 				const leqPartial = partialLengths.latestLeq(seq);
-				if (leqPartial) {
-					if (leqPartial.seq === seq) {
-						seqSeglen += leqPartial.seglen;
-					}
+				if (leqPartial && leqPartial.seq === seq) {
+					seqSeglen += leqPartial.seglen;
+					remoteObliteratedLen += leqPartial.remoteObliteratedLen ?? 0;
 				}
 				segCount += branchPartialLengths.segmentCount;
 			} else {
 				const segment = child;
 				const removalInfo = toRemovalInfo(segment);
-
-				if (segment.seq === seq) {
-					if (removalInfo?.removedSeq !== seq) {
+				const moveInfo = toMoveInfo(segment);
+
+				const removalIsLocal =
+					!!removalInfo && removalInfo.removedSeq === UnassignedSequenceNumber;
+				const moveIsLocal = !!moveInfo && moveInfo.movedSeq === UnassignedSequenceNumber;
+
+				const removeHappenedFirst =
+					removalInfo &&
+					(!moveInfo ||
+						moveIsLocal ||
+						(!removalIsLocal && moveInfo.movedSeq > removalInfo.removedSeq));
+
+				// eslint-disable-next-line unicorn/prefer-switch
+				if (seq === segment.seq) {
+					if (
+						segment.wasMovedOnInsert &&
+						segment.seq !== undefined &&
+						moveInfo &&
+						moveInfo.movedSeq < segment.seq
+					) {
+						remoteObliteratedLen += segment.cachedLength;
+					} else {
 						seqSeglen += segment.cachedLength;
 					}
-				} else {
-					if (removalInfo?.removedSeq === seq) {
+				} else if (seq === removalInfo?.removedSeq) {
+					if (removeHappenedFirst) {
+						seqSeglen -= segment.cachedLength;
+					} else {
+						remoteObliteratedLen -= segment.cachedLength;
+					}
+				} else if (seq === moveInfo?.movedSeq) {
```

Comment:
probably just brain fatigue, but when does this case happen? it clearly does as I see code coverage reports the lines as such :)

Reply:
I'll add some comments, since the logic is pretty dense